engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -460,4 +261,258 @@ Type getType() {
   public String toString() {
     return "Scope" + module;
   }
+
+  public static class Builder {
+
+    @CompilerDirectives.CompilationFinal private ModuleScope moduleScope = null;
+    private final Module module;
+    private final Type associatedType;
+    private Map<String, Object> polyglotSymbols;
```

Comment:
nitpick: fields `polyglotSymbols`, `type`, `methods`, `conversions`, `imports` and `exports` can all be final